sabnzbd/notifier.py

```diff
@@ -265,6 +285,92 @@ def send_prowl(title, msg, notification_type, force=False, test=None):
     return ""
 
 
+def send_apprise(title, msg, notification_type, force=False, test=None):
+    """send apprise message"""
+    logging.debug("Sending Apprise notification")
+    if test:
+        urls = test.get("apprise_urls")
+    else:
+        urls = sabnzbd.cfg.apprise_urls()
+
+    # Notification mapper
+    n_map = {
+        # Startup/Shutdown
+        "startup": apprise.common.NotifyType.INFO,
+        # Pause/Resume
+        "pause_resume": apprise.common.NotifyType.INFO,
+        # Added NZB
+        "download": apprise.common.NotifyType.INFO,
+        # Post-processing started
+        "pp": apprise.common.NotifyType.INFO,
+        # Job finished
+        "complete": apprise.common.NotifyType.SUCCESS,
+        # Job failed
+        "failed": apprise.common.NotifyType.FAILURE,
+        # Warning
+        "warning": apprise.common.NotifyType.WARNING,
+        # Error
+        "error": apprise.common.NotifyType.FAILURE,
+        # Disk full
+        "disk_full": apprise.common.NotifyType.WARNING,
+        # Queue finished
+        "queue_done": apprise.common.NotifyType.INFO,
+        # User logged in
+        "new_login": apprise.common.NotifyType.INFO,
+        # Other Messages
+        "other": apprise.common.NotifyType.INFO,
+    }
+
+    # Prepare our Asset Object
+    asset = apprise.AppriseAsset(
+        app_id="SABnzbd",
+        app_desc="SABnzbd Notification",
+        app_url="https://sabnzbd.org/",
+        image_path_mask=os.path.join(os.path.dirname(__file__), "apprise", "apprise-{TYPE}.png"),
+        image_url_mask="https://raw.githubusercontent.com/sabnzbd/sabnzbd.github.io/master/images/icons/apprise/{TYPE}.png",
+        image_url_logo="https://raw.githubusercontent.com/sabnzbd/sabnzbd.github.io/master/images/icons/"
+        "apple-touch-icon-180x180-precomposed.png",
+    )
+
+    # Initialize our Apprise Instance
+    apobj = apprise.Apprise(asset=asset)
+
+    if not test:
+        # Get a list of tags that are set to use the common list
+        target = get_target(notification_type, "apprise")
+        if isinstance(target, str):
+            if target:
+                # Store our URL and assign our key
+                if not apobj.add(target):
+                    logging.warning(
+                        "Key: %s - %s", notification_type, T("One or more Apprise URLs could not be loaded.")
+                    )
+
+            else:
+                # Use default list
+                apobj.add(urls)
+
+        else:
+            # Nothing to notify
+            return ""
+
+    else:
+        # Use default list
+        apobj.add(urls)
+
+    try:
+        # The below notifies anything added to our list
+        return (
```

Comment:
No need to use 1 liner here, let's just split it.

Reply:
Done